### DISCUSSION TITLE
Midi Tuning Standard : tone different from regular midi when playing the same note

### DISCUSSION THREAD

**Discussion by max22-** [FluidSynth/fluidsynth#865] [Q&A] [answered]
Hi,
i am working on a hobby project using libfluidsynth, to make an electronic "biniou", which are bagpipes from my region.
But since I use Midi Tuning Standard instead of "vanilla" midi, it doesn't sound the same. The pitches are correct, but the tone is not as good as before. It sounds a bit like libfluidsynth does not use the same samples as before to play the same note. (but maybe i am wrong)

Here is how my thing works :
Before :
- bapipes tuned in Bb major :
fingering "1111110" --> it is a Bb (midi number 82) --> note on 82
- bagpipes tuned in G major :
fingering "1111110" --> it is a G (midi number 79) -> note on 79 
The tone is ok for both cases

And now, with MTS :
- bagpipes tuned in Bb major :
fingering "1111110" --> it is a Bb (midi number 82, mapped to a Bb with midi fluid_synth_activate_key_tuning) --> note on 82
the tone is still OK 
- bagpipes tuned in G Major :
fingering "1111110" --> it is a Bb (not true but who cares ^^, it is mapped to a G) --> note on 82
the tone is not good
fingering "1000000" -> it is a G (not true but who cares ^^, it is mapped to an E) --> note on 88
the tone is not good at all, whereas it should sound good because there is an Eb sample in the soundfont.

It is a bit like fluidsynth didn't use the same samples as before to play the same note.

Should I use midi numbers close to the real notes to solve this problem ?
For example, if i want to play an A + 1/4 tone, should i assign it to an A, or can i assign it to whatever midi number i want, without altering the tone ?

I hope that what i wrote is not to chaotic ^^
And thanks for your wonderful library, i have a lot of fun with it !

**Comment by mawe42** [comment, marked as answer]
> Should I use midi numbers close to the real notes to solve this problem ?

Yes, that is the solution to your problem. The sample that is later pitch-shifted by MTS is *only* chosen by the note-number. FluidSynth doesn't look at the actual frequency to select a base good sample.

Great to hear that you are implementing an electronic Biniou, btw! I'm using FluidSynth for an electronic hurdy-gurdy :-)

**Reply by max22-** [reply]
Ok, thank you very much for your help !
you are the creator of the midigurdy ? i've seen some videos of it, very impressive !
I have this song in my playlists for when I wake up in the morning : https://www.youtube.com/watch?v=HcZOnlbn5fM ^^
the case looks very nice ! is it injection molded ?

**Reply by mawe42** [reply]
Yes, the MidiGurdy. Thanks for the praise :-) And I will let that customer know that you have his space impro on your morning playlist, he will like that a lot :-)

The body is milled from a solid block of polyurethane, then painted. But I'm planning a new version, simpler, easier to build, hopefully cheaper too.

Looking forward to your creation!
